spec/core_ext/hash_with_indifferent_access_spec.rb

```diff
@@ -19,6 +19,14 @@
     expect(@hash.fetch("foo", nil)).to eq("bar")
     expect(@hash.fetch(:foo)).to eq("bar")
     expect(@hash.fetch(:foo, nil)).to eq("bar")
+
+    expect(@hash.fetch("baz")).to eq("bee")
+    expect(@hash.fetch("baz", nil)).to eq("bee")
+    expect(@hash.fetch(:baz)).to eq("bee")
+    expect(@hash.fetch(:baz, nil)).to eq("bee")
+
+    expect { @hash.fetch(:missing) }.to raise_error(KeyError)
```

Comment:
The error raised is also an `IndexError`. 

https://stackoverflow.com/questions/17773687/hash-fetchnot-exist-key-raises-indexerror-instead-of-keyerror

```
irb(main):016:0> k=KeyError.exception("as")
#<KeyError: as>
irb(main):017:0> i=IndexError.exception("adsad")
#<IndexError: adsad>
irb(main):021:0> i.class.ancestors
[
    [0] IndexError < StandardError,
    [1] StandardError < Exception,
    [2] Exception < Object,
    [3] Object < BasicObject,
    [4] Kernel,
    [5] BasicObject
]
irb(main):022:0> k.class.ancestors
[
    [0] KeyError < IndexError,
    [1] IndexError < StandardError,
    [2] StandardError < Exception,
    [3] Exception < Object,
    [4] Object < BasicObject,
    [5] Kernel,
    [6] BasicObject
]
```